server/app/services/scheduler/strategy/random.rb

```diff
@@ -13,22 +13,32 @@ def host_grace_period
         30.seconds
       end
 
-      ##
       # @param [GridService] grid_service
       # @param [Integer] instance_number
-      # @param [Array<HostNode>] nodes
+      # @param [Array<Scheduler::Node>] nodes
+      # @return [Scheduler::Node, NilClass]
       def find_node(grid_service, instance_number, nodes)
         if grid_service.stateless?
+          node = find_previous_or_new_node(grid_service, instance_number, nodes)
+          return node if node
```

Comment:
Turns out that this matches the existing Kontena 1.1 behavior for stateful `ha` services, so I suppose this is working as intended.